### PR TITLE
feat: use-http-client

### PR DESCRIPTION
**Pull request by kraikov**
Replace all `axios` calls with  [HttpClient](https://github.com/liquality/chainify/blob/dev/packages/client/lib/Http.ts#L6)

### CODE REVIEW

File: src/store/utils.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export async function getPrices(baseCurrencies: string[], toCurrency: string) {
   const coindIds = baseCurrencies
     .filter((currency) => cryptoassets[currency]?.coinGeckoId)
     .map((currency) => cryptoassets[currency].coinGeckoId);
-  const { data } = await axios.get(
+  const data = await HttpClient.get(
```
Comment (adilmezghouti) [inline src/store/utils.ts:94]: [Boulder] What happens if the call fails?

Comment (adilmezghouti) [reply src/store/utils.ts:94]: [Pebble] Is there a way to cache the result for sometime?

File: src/swaps/liquality/LiqualitySwapProvider.ts [hunk 2]
```diff
@@ -77,15 +76,16 @@ export interface LiqualitySwapProviderConfig extends EvmSwapProviderConfig {
...
+
   private async getMarketInfo(): Promise<LiqualityMarketData[]> {
-    return (
-      await axios({
-        url: this.config.agent + '/api/swap/marketinfo',
-        method: 'get',
-        headers,
-      })
-    ).data;
+    return this._httpClient.nodeGet('/api/swap/marketinfo', null, { headers });
```
Comment (adilmezghouti) [inline src/swaps/liquality/LiqualitySwapProvider.ts:88]: [Boulder] Don't we need an await here?

File: src/swaps/liquality/LiqualitySwapProvider.ts [hunk 3]
```diff
@@ -245,18 +245,16 @@ export class LiqualitySwapProvider extends EvmSwapProvider {
   }
 
   public async updateOrder(order: LiqualitySwapHistoryItem) {
```
Comment (adilmezghouti) [inline src/swaps/liquality/LiqualitySwapProvider.ts:247]: [Pebble] Is it possible to add typescript types here?

File: src/swaps/liquality/LiqualitySwapProvider.ts [hunk 4]
```diff
@@ -451,14 +449,7 @@ export class LiqualitySwapProvider extends EvmSwapProvider {
...
     try {
-      return (
-        await axios({
-          url: this.config.agent + '/api/swap/order',
-          method: 'post',
-          data: { from, to, fromAmount: amount },
-          headers,
-        })
-      ).data;
+      return this._httpClient.nodePost('/api/swap/order', { from, to, fromAmount: amount }, { headers });
```
Comment (adilmezghouti) [inline src/swaps/liquality/LiqualitySwapProvider.ts:452]: [Boulder] Same comment about a missing await

File: src/swaps/oneinch/OneinchSwapProvider.ts [hunk 5]
```diff
@@ -70,21 +75,17 @@ class OneinchSwapProvider extends SwapProvider {
...
-    return await axios({
-      url: this.config.agent + `/${chainIdFrom}/quote`,
-      method: 'get',
-      params: {
-        fromTokenAddress: fromToken || NATIVE_ASSET_ADDRESS,
-        toTokenAddress: toToken || NATIVE_ASSET_ADDRESS,
-        amount,
-        fee,
-      },
+    return await this._httpClient.nodeGet(`/${chainIdFrom}/quote`, {
```
Comment (adilmezghouti) [inline src/swaps/oneinch/OneinchSwapProvider.ts:84]: [Boulder] How are we handling errors if they ever occur?

File: src/swaps/oneinch/OneinchSwapProvider.ts [hunk 6]
```diff
@@ -98,7 +99,7 @@ class OneinchSwapProvider extends SwapProvider {
     if (chainIdFrom !== chainIdTo || !chainToRpcProviders[chainIdFrom]) return null;
 
     const trade = await this._getQuote(chainIdFrom, from, to, fromAmountInUnit.toNumber());
-    const toAmountInUnit = new BN(trade.data.toTokenAmount);
+    const toAmountInUnit = new BN(trade?.toTokenAmount);
```
Comment (adilmezghouti) [inline src/swaps/oneinch/OneinchSwapProvider.ts:102]: [Boulder] What are the consequences in terms of logic if trade is falsy?

File: src/swaps/oneinch/OneinchSwapProvider.ts [hunk 7]
```diff
@@ -177,23 +174,19 @@ class OneinchSwapProvider extends SwapProvider {
...
-    if (new BN(quote.toAmount).times(1 - swapParams.slippage / 100).gt(trade.data.toTokenAmount)) {
+    if (new BN(quote.toAmount).times(1 - swapParams.slippage / 100).gt(trade?.toTokenAmount)) {
       throw new Error(
-        `Slippage is too high. You expect ${quote.toAmount} but you are going to receive ${trade.data.toTokenAmount} ${quote.to}`
+        `Slippage is too high. You expect ${quote.toAmount} but you are going to receive ${trade?.toTokenAmount} ${quote.to}`
       );
     }
 
     await this.sendLedgerNotification(quote.fromAccountId, 'Signing required to complete the swap.');
     const swapTx = await client.wallet.sendTransaction({
```
Comment (adilmezghouti) [inline src/swaps/oneinch/OneinchSwapProvider.ts:186]: [Boulder] Would it cause any issues if we submit a transaction with empty/null/undefined values?

File: src/utils/asset.ts [hunk 8]
```diff
@@ -268,7 +268,7 @@ export const estimateGas = async ({ data, to, value }: { data: string; to: strin
 export const fetchTerraToken = async (address: string) => {
   const {
     data: { mainnet: tokens },
-  } = await axios.get('https://assets.terra.money/cw20/tokens.json');
+  } = await HttpClient.get('https://assets.terra.money/cw20/tokens.json');
```
Comment (adilmezghouti) [inline src/utils/asset.ts:271]: [Pebble] Can we store this within the project if it does not change?

**Comment by kraikov** [comment]
@adilmezghouti I agree with all of your comments, but this is not the purpose of the PR. I haven't changed any logic.
The purpose is to replace all `axios` calls with [HttpClient](https://github.com/liquality/chainify/blob/dev/packages/client/lib/Http.ts#L6)

All of your comments could/should be addressed in another PR.

**Review by adilmezghouti — APPROVED** [review]

**Review by skeremidchiev — APPROVED** [review]

**Review by monokh — APPROVED** [review]

**Pull request merged**